### ISSUE TITLE
Crash on Treasure Map Read

### ISSUE THREAD

**Issue by Homicydal** [anegostudios/VintageStory-Issues#2691]
### Game Version

v1.18.1

### Platform

Windows

### Modded

None

### SP/MP

Multiplayer

### Description

I'm getting a nullref when I try to read the treasure map:

```
30.4.2023 22:48:25 [Server Warning] Exception at client 52. Disconnecting client.
30.4.2023 22:48:25 [Server Event] Player Homicydal got removed. Reason: Threw an exception at the server
30.4.2023 22:48:25 [Server Error] System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.ModSystemStructureLocator.GetStructure(StructureLocation loc) in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\LocatorMaps\StructureLocator.cs:line 33
   at Vintagestory.GameContent.ItemLocatorMap.getStructureCenter(ITreeAttribute attr) in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\LocatorMaps\ItemLocatorMap.cs:line 149
   at Vintagestory.GameContent.ItemLocatorMap.OnHeldInteractStart(ItemSlot slot, EntityAgent byEntity, BlockSelection blockSel, EntitySelection entitySel, Boolean firstEvent, EnumHandHandling& handling) in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\LocatorMaps\ItemLocatorMap.cs:line 84
   at Vintagestory.Server.ServerSystemInventory.HandleHandInteraction(Packet_Client packet, ConnectedClient client) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\Systems\Inventory.cs:line 253
   at Vintagestory.Server.ServerMain.ProcessNetMessage(NetIncomingMessage msg, NetServer mainSocket) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\ServerMainNetworking.cs:line 107
```

This world was created with an rc version prior to going to stable, is there any way to resolve this?

### How to reproduce

Right click on the treasure map

### Logs

```Log```

**Comment by radfast** [comment]
Thanks for reporting, should be fixed in upcoming 1.18.2

**Comment by Homicydal** [comment]
Thank you, I ran `1.18.2rc4` and it does appear that the crash is resolved, but the treasure map reports "No location found on this map", is this expected?
